hack/uwm_setup.sh

```diff
@@ -0,0 +1,259 @@
+#  LATER) TEST ENVIRONMENTS ONLY. THE SCRIPT MAKES CHANGES TO OAUTH IDENTITY
+# PROVIDER CONFIGURATION, OPENSHIFT USER ACCOUNTS AND USER WORKLOAD MONITORING
+#CONFIGURATION. THE SCRIPT ALSO DEPLOYS A USER APPLICATION IN A NAMESPACE CALLED
+#  'ns1'. REVIEW THIS SCRIPT THOROUGHLY TO DETERMINE SUITABILITY FOR YOUR TEST
+#                                  ENVIRONMENT
+#*******************************************************************************
+#
+# This script automates user workload monitoring setup for newly deployed OpenShift Container Platform 4.6 (or later) test clusters.The script does the following:
+#
+# * Enables user workload monitoring
+# * Deploys a user application
+# * Configures HTPasswd identity provider, creates a secret and creates some users
+# * Applies roles to those users
+# * Adds a service monitor
+# * Creates an alerting rule
+# * Outputs cluster and Thanos URLs to stdout
+#
+# The script assumes that YAML files `example-app-alerting-rule.yaml`, `example-app-service-monitor.yaml` and `prometheus-example-app.yaml` are present in the current working directory along with the script itself.
+
+# Add a trap which removes temporary files on SIGHUP, SIGTERM, SIGKILL, SIGQUIT, SIGINT and EXIT. Commented out by default. Review for suitability in your enviornment before uncommenting:
+# trap 'rm ./tmp/*' SIGHUP SIGTERM SIGKILL SIGQUIT SIGINT EXIT
+
+if [ ! -d "./tmp" ]; then
+  mkdir ./tmp
+fi
+
+function notice ()
+{
+echo "*******************************************************************************
+  NOTE THAT THIS SCRIPT IS INTENDED FOR OPENSHIFT CONTAINER PLATFORM 4.6 (OR
+  LATER) TEST ENVIRONMENTS ONLY. THE SCRIPT MAKES CHANGES TO OAUTH IDENTITY
+ PROVIDER CONFIGURATION, OPENSHIFT USER ACCOUNTS AND USER WORKLOAD MONITORING
+CONFIGURATION. THE SCRIPT ALSO DEPLOYS A USER APPLICATION IN A NAMESPACE CALLED
+  'ns1'. REVIEW THIS SCRIPT THOROUGHLY TO DETERMINE SUITABILITY FOR YOUR TEST
+                                  ENVIRONMENT
+*******************************************************************************"
+while true; do
+  read -rp "Do you want to continue? (y/n): " YESNO
+  case $YESNO in
+      y ) break;;
+      n ) echo "Exiting."; exit 0;;
+      * ) echo "Please answer 'y' or 'n':";;
+  esac
+done
+echo ""
+}
+
+function kubeadmin_login ()
+{
+echo "Log into your cluster initially as kubeadmin:"; echo ""
+
+# Request cluster API endpoint and kubeadmin token from user:
+read -rp "- Enter cluster API endpoint and port (i.e. https://api.<cluster_name>.<sub_domain>.<domain>:<port>: " APIURL
+echo ""
+unset -v KUBEPW # Make sure the $KUBEPW password variable is not exported
+set +o allexport  # Make sure variables are not automatically exported
+read -rs -p "- Enter kubeadmin token (this will not be echoed to the console and the variable will not be exported): " KUBEPW < /dev/tty &&
+echo ""
+
+# Log in as kubeadmin:
+oc login --token="${KUBEPW}" --server="${APIURL}"
```

Comment:
Shellcheck is not happy about it. I think sth like this should do the trick:
```suggestion
oc login --token="${KUBEPW[0]}" --server="${APIURL[0]}"
```

Reply:
Thank you, I have now added this change to the script and that resolved the issue.

As mentioned just now in https://github.com/openshift/cluster-monitoring-operator/pull/849#issuecomment-657549022, shellcheck is failing on one line only now, because it does not seem to be picking up the double quote around the variable due to it being in a double quoted echo statement.